tests/lib/rules/require-expect.js

```diff
@@ -0,0 +1,195 @@
+                "    });",
+                "});"
+            ].join(returnAndIndent),
+            options: ["except-simple"]
+        }
+
+    ],
+
+    invalid: [
+        // default, make sure expect is identified correctly
+        {
+            code: "test('name', function(assert) { other.assert.expect(0) });",
+            errors: [{
+                message: EXCEPT_SIMPLE_ERROR_MESSAGE,
+                type: "CallExpression"
+            }],
+            options: []
+        },
+
+        // assert in loop block
+        {
+            code: "test('name', function(assert) { while (false) { assert.ok(true) } });",
+            errors: [{
+                message: ALWAYS_ERROR_MESSAGE,
+                type: "CallExpression"
+            }],
+            options: ["except-simple"]
+        },
+
+        // assert used in callback
+        {
+            code: "test('name', function(assert) { maybe(function() { assert.ok(true) }); });",
+            errors: [{
+                message: ALWAYS_ERROR_MESSAGE,
+                type: "CallExpression"
+            }],
+            options: ["except-simple"]
+        },
+
+        // assert in function expression
+        {
+            code: "test('name', function(assert) { var maybe = function() { assert.ok(true) }; });",
+            errors: [{
+                message: ALWAYS_ERROR_MESSAGE,
+                type: "CallExpression"
+            }],
+            options: ["except-simple"]
+        },
+
+        // `expect` does not count when used inside of a block.
+        {
+            code: "test('name', function(assert) { function name() { assert.expect(0); assert.ok(true) } });",
+            errors: [{
+                message: ALWAYS_ERROR_MESSAGE,
+                type: "CallExpression"
+            }],
+            options: ["except-simple"]
+        },
+
+        // assert in outer test context and nested in a block
+        {
+            code: "test('name', function(assert) { assert.ok(true); if (true) { assert.ok(true); } });",
+            errors: [{
+                message: ALWAYS_ERROR_MESSAGE,
+                type: "CallExpression"
+            }],
+            options: ["except-simple"]
+        },
+
+        // Deeply nested
+        {
+            code: "test('name', function(assert) { if (true) { if (true) { assert.ok(true); } } });",
+            errors: [{
+                message: ALWAYS_ERROR_MESSAGE,
+                type: "CallExpression"
+            }],
+            options: ["except-simple"]
+        },
+
+        // Sending assert to a function
+        {
+            code: [
+                "function myAssertion(a, assert, c) { assert.ok(true); }",
+                "test('name', function(assert) { myAssertion(null, assert, null); });"
+            ].join(returnAndIndent),
+            errors: [{
+                message: ALWAYS_ERROR_MESSAGE,
+                type: "CallExpression"
+            }],
+            options: ["except-simple"]
+        },
```

Comment:
Could you add a test case for this exact case, except that `assert` was renamed in the test function? (That is, `test('name', function(foo) { myAssertion(null, fo, null); });`) Thanks!